src/source/ObjcGenerator.scala

```diff
@@ -128,14 +130,14 @@ class ObjcGenerator(spec: Spec) extends BaseObjcGenerator(spec) {
     writeObjcFile(marshal.headerName(ident), origin, refs.header, w => {
       for (c <- i.consts if marshal.canBeConstVariable(c)) {
         writeDoc(w, c.doc)
-        w.w(s"extern ")
+        w.w(s"${visibilityAnnotation}extern ")
```

Comment:
I wonder if constants visibility should be declared in `BaseObjcGenerator.writeObjcConstMethImpl`?

Reply:
My understanding is that visibility attributes should be placed in declarations not definitions. Here's a quote from the GCC wiki (https://gcc.gnu.org/wiki/Visibility):
>In your header files, wherever you want an interface or API made public outside the current DSO, place `__attribute__ ((visibility ("default")))` in struct, class and function declarations you wish to make public (it's easier if you define a macro as this). You don't need to specify it in the definition.

Reply:
I see.  I was thinking for constants,  adding to `writeObjcConstMethImpl` will need only 1 change than the declarations where 2 changes are needed.

Reply:
maybe, but writeObjcConstVariableDecl has 3 callers,  2 with `extern`, 1 without.

Reply:
Right, so I'd have to add an argument to indicate whether the declaration is in the header (possibly add visibility) or the implementation (don't add visibility)? Not sure if it's worth it in this case, but I don't have strong feelings either way.